From e156ac781074af02d9a5ed4076a561f75418f255 Mon Sep 17 00:00:00 2001
From: Steven Downer <grinch@outlook.com>
Date: Tue, 1 Oct 2013 22:48:23 -0500
Subject: [PATCH] (Almura) Update POM.xml

- Change 'api.version' from 'unknown' to 'v1_6_R3'
- Change 'defaultGoal' from 'package' to 'clean package'
- Change 'shadedPattern' to remove versioned classpaths
- Change mapper input and output to match a static name
- Change 'target' and 'source' from Java 1.6 to Java 1.7
- Add 'commons-lang' dependency
- Add relocation for CraftServer for plugin compatibility checks
- Update 'maven-compiler-plugin' from '2.3.2' to '3.1'
- Update 'maven-shade-plugin' from '1.4' to '2.1'

diff --git a/dependency-reduced-pom.xml b/dependency-reduced-pom.xml
new file mode 100644
index 0000000..f314a91
--- /dev/null
+++ b/dependency-reduced-pom.xml
@@ -0,0 +1,258 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.almuradev</groupId>
+  <artifactId>almura-server</artifactId>
+  <name>Almura-Server</name>
+  <version>1.6.4-R2.1-SNAPSHOT</version>
+  <build>
+    <defaultGoal>clean package</defaultGoal>
+    <finalName>${project.name}</finalName>
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <groupId>org.eclipse.m2e</groupId>
+          <artifactId>lifecycle-mapping</artifactId>
+          <version>1.0.0</version>
+          <configuration>
+            <lifecycleMappingMetadata>
+              <pluginExecutions>
+                <pluginExecution>
+                  <pluginExecutionFilter>
+                    <groupId>com.lukegb.mojo</groupId>
+                    <artifactId>gitdescribe-maven-plugin</artifactId>
+                    <versionRange>[1.3,)</versionRange>
+                    <goals>
+                      <goal>gitdescribe</goal>
+                    </goals>
+                  </pluginExecutionFilter>
+                  <action>
+                    <ignore />
+                  </action>
+                </pluginExecution>
+              </pluginExecutions>
+            </lifecycleMappingMetadata>
+          </configuration>
+        </plugin>
+      </plugins>
+    </pluginManagement>
+    <plugins>
+      <plugin>
+        <groupId>com.lukegb.mojo</groupId>
+        <artifactId>gitdescribe-maven-plugin</artifactId>
+        <version>1.3</version>
+        <executions>
+          <execution>
+            <phase>compile</phase>
+            <goals>
+              <goal>gitdescribe</goal>
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <outputPrefix>git-Spigot-</outputPrefix>
+          <outputPostfix />
+        </configuration>
+      </plugin>
+      <plugin>
+        <artifactId>maven-jar-plugin</artifactId>
+        <version>2.1</version>
+        <configuration>
+          <archive>
+            <manifestEntries>
+              <Main-Class>org.bukkit.craftbukkit.Main</Main-Class>
+              <Implementation-Title>CraftBukkit</Implementation-Title>
+              <Implementation-Version>${describe}</Implementation-Version>
+              <Implementation-Vendor>Bukkit Team</Implementation-Vendor>
+              <Specification-Title>Bukkit</Specification-Title>
+              <Specification-Version>${api.version}</Specification-Version>
+              <Specification-Vendor>Bukkit Team</Specification-Vendor>
+              <Sealed>true</Sealed>
+            </manifestEntries>
+            <manifestSections>
+              <manifestSection>
+                <name>net/bukkit/</name>
+                <manifestEntries>
+                  <Sealed>true</Sealed>
+                </manifestEntries>
+              </manifestSection>
+              <manifestSection>
+                <name>com/bukkit/</name>
+                <manifestEntries>
+                  <Sealed>true</Sealed>
+                </manifestEntries>
+              </manifestSection>
+              <manifestSection>
+                <name>org/bukkit/</name>
+                <manifestEntries>
+                  <Sealed>true</Sealed>
+                </manifestEntries>
+              </manifestSection>
+            </manifestSections>
+          </archive>
+        </configuration>
+      </plugin>
+      <plugin>
+        <artifactId>maven-shade-plugin</artifactId>
+        <version>2.1</version>
+        <executions>
+          <execution>
+            <phase>package</phase>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+            <configuration>
+              <relocations>
+                <relocation>
+                  <pattern>org.bouncycastle</pattern>
+                  <shadedPattern>net.minecraft.org.bouncycastle</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>joptsimple</pattern>
+                  <shadedPattern>org.bukkit.craftbukkit.libs.joptsimple</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>jline</pattern>
+                  <shadedPattern>org.bukkit.craftbukkit.libs.jline</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.ibex</pattern>
+                  <shadedPattern>org.bukkit.craftbukkit.libs.org.ibex</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.gjt</pattern>
+                  <shadedPattern>org.bukkit.craftbukkit.libs.org.gjt</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>com.google.gson</pattern>
+                  <shadedPattern>org.bukkit.craftbukkit.libs.com.google.gson</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.objectweb.asm</pattern>
+                  <shadedPattern>com.almuradev.internal.libs.org.objectweb.asm</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.bukkit.craftbukkit</pattern>
+                  <shadedPattern>org.bukkit.craftbukkit.v${minecraft_version}</shadedPattern>
+                  <includes>
+                    <include>org.bukkit.craftbukkit.CraftServer*</include>
+                  </includes>
+                </relocation>
+              </relocations>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.1</version>
+        <configuration>
+          <source>1.7</source>
+          <target>1.7</target>
+          <encoding>${project.build.sourceEncoding}</encoding>
+        </configuration>
+      </plugin>
+      <plugin>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <version>2.11</version>
+        <configuration>
+          <workingDirectory>${basedir}/target/test-server</workingDirectory>
+          <excludes>
+            <exclude>org/bukkit/craftbukkit/updater/BukkitDLUpdaterServiceTest.java</exclude>
+            <exclude>org/bukkit/craftbukkit/inventory/ItemStack*Test.java</exclude>
+          </excludes>
+        </configuration>
+      </plugin>
+      <plugin>
+        <artifactId>maven-resources-plugin</artifactId>
+        <version>2.4.3</version>
+        <executions>
+          <execution>
+            <id>maps-file</id>
+            <phase>package</phase>
+            <goals>
+              <goal>copy-resources</goal>
+            </goals>
+            <configuration>
+              <outputDirectory>${basedir}/target</outputDirectory>
+              <resources>
+                <resource>
+                  <directory>${basedir}</directory>
+                  <filtering>true</filtering>
+                  <includes>
+                    <include>maps.yml</include>
+                  </includes>
+                </resource>
+              </resources>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>com.wolvereness</groupId>
+        <artifactId>overmapped</artifactId>
+        <version>0.0.2</version>
+        <executions>
+          <execution>
+            <phase>package</phase>
+            <goals>
+              <goal>map</goal>
+            </goals>
+            <configuration>
+              <maps>${basedir}/target/maps.yml</maps>
+              <input>${basedir}/target/${project.name}.jar</input>
+              <original>${basedir}/target/unmapped-${project.name}.jar</original>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+  <repositories>
+    <repository>
+      <id>repobo-snap</id>
+      <url>http://repo.bukkit.org/content/groups/public</url>
+    </repository>
+  </repositories>
+  <pluginRepositories>
+    <pluginRepository>
+      <id>bukkit-plugins</id>
+      <url>http://repo.bukkit.org/content/groups/public</url>
+    </pluginRepository>
+  </pluginRepositories>
+  <dependencies>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.11</version>
+      <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <artifactId>hamcrest-core</artifactId>
+          <groupId>org.hamcrest</groupId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.hamcrest</groupId>
+      <artifactId>hamcrest-library</artifactId>
+      <version>1.3</version>
+      <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <artifactId>hamcrest-core</artifactId>
+          <groupId>org.hamcrest</groupId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+  </dependencies>
+  <properties>
+    <minecraft_version>1_6_R3</minecraft_version>
+    <buildtag.prefix>git-Bukkit-</buildtag.prefix>
+    <api.version>v1_6_R3</api.version>
+    <minecraft.version>1.6.4</minecraft.version>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <junit.version>4.11</junit.version>
+  </properties>
+</project>
+
diff --git a/pom.xml b/pom.xml
index dc0a64b..ffd1530 100644
--- a/pom.xml
+++ b/pom.xml
@@ -2,23 +2,15 @@
   xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
 
-  <parent>
-    <groupId>org.spigotmc</groupId>
-    <artifactId>spigot-parent</artifactId>
-    <version>dev-SNAPSHOT</version>
-    <relativePath>../pom.xml</relativePath>
-  </parent>
-
-  <groupId>org.spigotmc</groupId>
-  <artifactId>spigot</artifactId>
+  <groupId>com.almuradev</groupId>
+  <artifactId>almura-server</artifactId>
   <packaging>jar</packaging>
   <version>1.6.4-R2.1-SNAPSHOT</version>
-  <name>Spigot</name>
-  <url>http://www.spigotmc.org</url>
+  <name>Almura-Server</name>
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <api.version>unknown</api.version>
+    <api.version>v1_6_R3</api.version>
     <junit.version>4.11</junit.version>
     <minecraft.version>1.6.4</minecraft.version>
     <minecraft_version>1_6_R3</minecraft_version>
@@ -42,8 +34,8 @@
 
   <dependencies>
     <dependency>
-      <groupId>org.spigotmc</groupId>
-      <artifactId>spigot-api</artifactId>
+      <groupId>com.almuradev</groupId>
+      <artifactId>almura-api</artifactId>
       <version>${project.version}</version>
       <type>jar</type>
       <scope>compile</scope>
@@ -96,6 +88,11 @@
       <artifactId>gson</artifactId>
       <version>2.1</version>
     </dependency>
+    <dependency>
+      <groupId>org.ow2.asm</groupId>
+      <artifactId>asm-commons</artifactId>
+      <version>4.0</version>
+    </dependency>
     <!-- testing -->
     <dependency>
       <groupId>junit</groupId>
@@ -133,7 +130,7 @@
 
   <!-- This builds a completely 'ready to start' jar with all dependencies inside -->
   <build>
-    <defaultGoal>install</defaultGoal>
+    <defaultGoal>clean package</defaultGoal>
     <plugins>
       <plugin>
         <groupId>com.lukegb.mojo</groupId>
@@ -194,7 +191,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-shade-plugin</artifactId>
-        <version>2.0</version>
+        <version>2.1</version>
         <executions>
           <execution>
             <phase>package</phase>
@@ -205,7 +202,7 @@
               <relocations>
                 <relocation>
                   <pattern>org.bouncycastle</pattern>
-                  <shadedPattern>net.minecraft.v${minecraft_version}.org.bouncycastle</shadedPattern>
+                  <shadedPattern>net.minecraft.org.bouncycastle</shadedPattern>
                 </relocation>
                 <relocation>
                   <pattern>joptsimple</pattern>
@@ -227,6 +224,10 @@
                   <pattern>com.google.gson</pattern>
                   <shadedPattern>org.bukkit.craftbukkit.libs.com.google.gson</shadedPattern>
                 </relocation>
+                <relocation>
+                  <pattern>org.objectweb.asm</pattern>
+                  <shadedPattern>com.almuradev.internal.libs.org.objectweb.asm</shadedPattern>
+                </relocation>
                 <!-- These contain native libraries, and cannot be properly relocated
                 <relocation>
                   <pattern>org.fusesource</pattern>
@@ -240,17 +241,25 @@
                   <pattern>com.mysql</pattern>
                   <shadedPattern>org.bukkit.craftbukkit.libs.${project.dependencies[6].version}.com.mysql</shadedPattern>
                 </relocation>
+                <relocation>
+                  <pattern>org.bukkit.craftbukkit</pattern>
+                  <shadedPattern>org.bukkit.craftbukkit.v${minecraft_version}</shadedPattern>
+                  <excludes>
+                    <exclude>org.bukkit.craftbukkit.Main*</exclude>
+                  </excludes>
+                </relocation>
+                <relocation>
+                  <pattern>net.minecraft.server</pattern>
+                  <shadedPattern>net.minecraft.server.v${minecraft_version}</shadedPattern>
+                </relocation>
                 !-->
+                <!--  Almura CraftServer relocation -->
                 <relocation>
                     <pattern>org.bukkit.craftbukkit</pattern>
                     <shadedPattern>org.bukkit.craftbukkit.v${minecraft_version}</shadedPattern>
-                    <excludes>
-                        <exclude>org.bukkit.craftbukkit.Main*</exclude>
-                    </excludes>
-                </relocation>
-                <relocation>
-                    <pattern>net.minecraft.server</pattern>
-                    <shadedPattern>net.minecraft.server.v${minecraft_version}</shadedPattern>
+                    <includes>
+                        <include>org.bukkit.craftbukkit.CraftServer*</include>
+                    </includes>
                 </relocation>
               </relocations>
             </configuration>
@@ -260,10 +269,11 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <version>2.3.2</version>
+        <version>3.1</version>
         <configuration>
-          <source>1.6</source>
-          <target>1.6</target>
+          <source>1.7</source>
+          <target>1.7</target>
+          <encoding>${project.build.sourceEncoding}</encoding>
         </configuration>
       </plugin>
       <plugin>
@@ -315,8 +325,8 @@
             </goals>
             <configuration>
               <maps>${basedir}/target/maps.yml</maps>
-              <input>${basedir}/target/${project.artifactId}-${project.version}.jar</input>
-              <original>${basedir}/target/unmapped-${project.artifactId}-${project.version}.jar</original>
+              <input>${basedir}/target/${project.name}-${project.version}.jar</input>
+              <original>${basedir}/target/unmapped-${project.name}-${project.version}.jar</original>
             </configuration>
           </execution>
         </executions>
-- 
1.9.0.msysgit.0

